### PR TITLE
Change the structure of directory and add main functions

### LINKED ISSUE #2: Make basic directories divided by functions
Make source directories such as Master, Worker, Network, etc.

### PR DESCRIPTION
**Pull request by JeongHunP**
fix #2, #4 

1. root(.) main에서 시작해서 args에 따라 master/worker main을 실행.
2. master(/master) - master.scala 안에 현재 server랑 main 합쳐져있음
worker(/worker) - worker.scala 안에 현재 client랑 main 합쳐져있음

여기서 각각 master worker 구동돼서 서버랑 클라이언트 만들어지고 소통하는거 까지 확인

ex) 332Project에서 sbt "run master 2" -> master main 실행해서 server 열기
sbt "run worker 127.0.1.1:50051" -> worker main 실행해서 client 열기

### CODE REVIEW

File: build.sbt [hunk 1]
```diff
@@ -11,31 +11,45 @@ lazy val settings = Seq(
   )
 )
 
+lazy val commonAssemblySettings = Seq(
```
Comment (daehuikim) [inline build.sbt:14]: What is the role of this libraries?

File: master/src/main/scala/master.scala [hunk 2]
```diff
@@ -1,7 +1,108 @@
...
+object MasterWorkerServer {
+  private val logger = Logger.getLogger(classOf[MasterWorkerServer].getName)
+
   def main(args: Array[String]): Unit = {
-    println("Hello")
+    // should be args len 1 which is master 3(num of workers)
+    require(args.length == 1 && args(0).toInt >= 1 && args(0).toInt <= 10)
+    val num_workers = args(0)
+
+    println(InetAddress.getLocalHost.getHostAddress)
```
Comment (daehuikim) [inline master/src/main/scala/master.scala]: Can we change this line with logging?

File: src/main/scala/Main.scala [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+  def main(args: Array[String]): Unit = {
+    require(args.length >= 2 && (args(0) == "master" || args(0) == "worker"))
+    val machineType = args(0)
+    val newArgs = args.slice(1, args.length)
+    if (machineType == "master") {
+      MasterWorkerServer.main(newArgs)
+    } else if (machineType == "worker") {
+      MasterWorkerClient.main(newArgs)
+    } else {
+      println("Illegal Arguments")
```
Comment (daehuikim) [inline src/main/scala/Main.scala]: Sub with log?

**Review by daehuikim — COMMENTED** [review]
* Which logger will we use?
* Worker command on Main.scala?

File: build.sbt (hunk 1 above)
Comment (JeongHunP) [reply build.sbt:14]: It is stated in here.
https://github.com/sbt/sbt-assembly

File: src/main/scala/Main.scala (hunk 3 above)
Comment (JeongHunP) [reply src/main/scala/Main.scala]: fixed

File: master/src/main/scala/master.scala (hunk 2 above)
Comment (JeongHunP) [reply master/src/main/scala/master.scala]: fixed

**Review by daehuikim — APPROVED** [review]

**Pull request merged**